server/src/jump_definition.rs

```diff
@@ -0,0 +1,182 @@
+}
+
+impl JumpVisitor {
+    pub fn new(search_location: Location) -> Self {
+        JumpVisitor {
+            search_location,
+            found_span: None,
+        }
+    }
+
+    fn check_comment(&mut self, commentable: &dyn Commentable) {
+        if let Some(comment) = commentable.comment() {
+            comment
+                .overview
+                .as_ref()
+                .map(|overview| self.check_message_links(&overview.message));
+            comment
+                .returns
+                .iter()
+                .for_each(|returns| self.check_message_links(&returns.message));
+            comment
+                .params
+                .iter()
+                .for_each(|params| self.check_message_links(&params.message));
+            comment
+                .throws
+                .iter()
+                .for_each(|throws| self.check_message_links(&throws.message));
+            comment
+                .throws
+                .iter()
+                .for_each(|s| self.check_and_set_span(s.thrown_type(), &s.span));
+            comment
+                .see
+                .iter()
+                .for_each(|s| self.check_and_set_span(s.linked_entity(), &s.span));
+        }
+    }
+
+    fn check_message_links(&mut self, message: &Vec<MessageComponent>) {
+        message.iter().for_each(|m| {
+            if let MessageComponent::Link(l) = m {
+                self.check_and_set_span(l.linked_entity(), &l.span);
+            }
+        });
+    }
+
+    fn check_and_set_span<T: Entity + ?Sized>(
+        &mut self,
+        linked_entity_result: Result<&T, &Identifier>,
+        span: &Span,
+    ) {
+        if let Ok(entity) = linked_entity_result {
+            if self.search_location.is_within(span) {
+                self.found_span = Some(entity.raw_identifier().span.clone());
+            };
+        }
+    }
+}
+
+impl Visitor for JumpVisitor {
+    fn visit_file(&mut self, _: &SliceFile) {}
+
+    fn visit_module(&mut self, _: &Module) {}
+
+    fn visit_struct(&mut self, struct_def: &Struct) {
+        self.check_comment(struct_def);
+    }
+
+    fn visit_class(&mut self, class_def: &Class) {
+        self.check_comment(class_def);
+        if let Some(base_ref) = &class_def.base {
+            if self.search_location.is_within(&base_ref.span) {
+                self.found_span = Some(base_ref.definition().raw_identifier().span.clone());
+            }
+        }
+    }
+
+    fn visit_exception(&mut self, exception_def: &Exception) {
+        self.check_comment(exception_def);
+        if let Some(base_ref) = &exception_def.base {
+            if self.search_location.is_within(&base_ref.span) {
+                self.found_span = Some(base_ref.definition().raw_identifier().span.clone());
+            }
+        }
+    }
+
+    fn visit_interface(&mut self, interface_def: &Interface) {
+        self.check_comment(interface_def);
+        interface_def.bases.iter().for_each(|base_ref| {
```

Comment:
Another general comment, using for loops is more conventional than `for_each` when possible.
Generally, it's easier to read, and better performant, since it doesn't involve closures.

Also semantically, it's a little different. `for_each` is actually returning a value from this function, which is kind of weird since it doesn't have a return type. Whereas for loops don't return anything. This compiles because the compiler infers an implicit return type of `-> ()` for this function, but still.